cli/cmd/upgrade.go

```diff
@@ -93,42 +109,37 @@ and apply any required migrations and adaptations.`,
 				}
 			}
 		} else {
-			operation = "Focefully upgrading"
-		}
-
-		config, err := resources.GetCurrentConfig(ctx, client, ns)
-		if err != nil {
-			fmt.Println("Odigos upgrade failed - unable to read the current Odigos configuration.")
-			os.Exit(1)
-		}
-
-		// update the config on upgrade
-		config.ConfigVersion += 1
-		if uiMode != "" {
-			config.UiMode = common.UiMode(uiMode)
+			operation = "Forcefully upgrading"
 		}
 
-		// Migrate images from prior to registry.odigos.io
-		if config.ImagePrefix == "" {
-			config.ImagePrefix = k8sconsts.OdigosImagePrefix
-		}
+		onPremTokenSet := cmd.Flag("onprem-token").Changed
+		var currentTier common.OdigosTier
 
-		currentTier, err := odigospro.GetCurrentOdigosTier(ctx, client, ns)
-		if err != nil {
-			fmt.Println("Odigos cloud login failed - unable to read the current Odigos tier.")
-			os.Exit(1)
+		if onPremTokenSet {
+			currentTier = common.OnPremOdigosTier
+		} else {
+			currentTier, err = odigospro.GetCurrentOdigosTier(ctx, client, ns)
+			if err != nil {
+				fmt.Println("Odigos cloud login failed - unable to read the current Odigos tier.")
+				os.Exit(1)
+			}
 		}
 
 		managerOpts := resourcemanager.ManagerOpts{
 			ImageReferences: GetImageReferences(currentTier, openshiftEnabled),
 		}
 
-		resourceManagers := resources.CreateResourceManagers(client, ns, currentTier, nil, config, versionFlag, installationmethod.K8sInstallationMethodOdigosCli, managerOpts)
+		resourceManagers := resources.CreateResourceManagers(
+			client, ns, currentTier, &odigosOnPremToken, config, versionFlag,
```

Comment:
before this PR we would only call `CreateResourceManagers` with `onPremToken` equals `nil` to indicate we don't want to modify it.

now we will always set the 4th parameter to `&odigosOnPremToken` to possible empty string. this will be changed for `community` tiers as well, and also for `odigos update` without a new token.

Please follow the code to make sure this change will not affect any other command. specifically, I am worried about [this section](https://github.com/odigos-io/odigos/blob/fd56e0f070c1e29d63b8c3667028646fe1752208/cli/cmd/resources/odigospro/manager.go#L35) which checks for nil but not for empty strings